### ISSUE TITLE
ServiceWorkerModule breaks App on Safari

### ISSUE THREAD

**Issue by maciejtreder** [angular/angular#20519]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior
Angular application becomes unusable on Safari when ServiceWorkerModule is used. Several exceptions are thrown:

1. Problem with SwPush, described in: #20407
2. Problem with SwUpdate:

```
[Error] ERROR – Error: Service workers are not supported by this browser — service-worker.js:47
Error: Service workers are not supported by this browser — service-worker.js:47
	error
	View_AppComponent_Host_0 (AppComponent_Host.ngfactory.js:4)
	logError (vendor.dll.js:15347)
	(anonymous function)
	handleError (vendor.dll.js:2078)
	run (client.js:10906)
	tick (vendor.dll.js:6460)
	_loadComponent (vendor.dll.js:6519)
	bootstrap (vendor.dll.js:6407)
	forEach
	_moduleDoBootstrap (vendor.dll.js:6127)
	(anonymous function) (vendor.dll.js:6048)
	onInvoke (vendor.dll.js:5296)
	run (client.js:10906)
	(anonymous function) (client.js:11637)
	onInvokeTask (vendor.dll.js:5287)
	runTask (client.js:10956)
	drainMicroTaskQueue (client.js:11366)
	promiseReactionJob
[Error] ERROR CONTEXT – DebugContext_ {view: Object, nodeIndex: 0, nodeDef: Object, …}
DebugContext_ {view: Object, nodeIndex: 0, nodeDef: Object, elDef: Object, elView: Object, …}DebugContext_
	error
	View_AppComponent_Host_0 (AppComponent_Host.ngfactory.js:4)
	logError (vendor.dll.js:15347)
	(anonymous function)
	handleError (vendor.dll.js:2083)
	run (client.js:10906)
	tick (vendor.dll.js:6460)
	_loadComponent (vendor.dll.js:6519)
	bootstrap (vendor.dll.js:6407)
	forEach
	_moduleDoBootstrap (vendor.dll.js:6127)
	(anonymous function) (vendor.dll.js:6048)
	onInvoke (vendor.dll.js:5296)
	run (client.js:10906)
	(anonymous function) (client.js:11637)
	onInvokeTask (vendor.dll.js:5287)
	runTask (client.js:10956)
	drainMicroTaskQueue (client.js:11366)
	promiseReactionJob
```

## Expected behavior
ServiceWorkerModule should check if browsers support service worker, instead of throwing errors.

## What is the motivation / use case for changing the behavior?
cross-browser support

## Environment

<pre><code>
Angular version: 5.0.2

Browser:
- [ ] Chrome (desktop) version XX
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [x] Safari (desktop) version XX
- [x] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX

</code></pre>

**Comment by MohammedSiddiqui10p** [comment]
From what I recall, I believe it's our responsibility to feature detect whether or not a browser can support a service worker or not.
```javascript
platformBrowserDynamic().bootstrapModule(AppModule).then(() => {
 if ('serviceWorker' in navigator) {
  navigator.serviceWorker.register('/route-to-worker.js')
 }
})
```

This may help:
[Great Progressive Web App Experiences with Angular (Google I/O '17)](https://youtu.be/C8KcW1Nj3Mw?t=20m)

**Comment by alxhub** [comment]
Fixed by b9a91a5e7490d499dcc4c698b4201962a01aff77.
